ModelLibrary/ModelAD/MAppointmentsInfo.cs

```diff
@@ -68,28 +68,39 @@ protected override bool AfterSave(bool newRecord, bool success)
             if (!success)
                 return success;
 
-            if (newRecord)
+            #region Push Notification
+
+            string type, title, body;
+
+            if (IsTask())
             {
-                string type, title, body;
-                if (IsTask())
-                {
-                    string priorityValue = $@"SELECT Name FROM AD_Ref_List WHERE AD_Reference_ID = 
+                string priorityValue = $@"SELECT Name FROM AD_Ref_List WHERE AD_Reference_ID = 
                                             (SELECT AD_Reference_ID FROM AD_Reference WHERE Name = '_PriorityRule')
                                             AND Value = {GetPriorityKey()} AND IsActive = 'Y'";
-                    string priority = Util.GetValueOfString(DB.ExecuteScalar(priorityValue));
-                    type = "T";
-                    title = Msg.GetMsg(GetCtx(), "Task");
-                    body = GetSubject() + " (" + priority + ")";
-                }
-                else
-                {
-                    type = "A";
-                    title = Msg.GetMsg(GetCtx(), "Appointment");
-                    body = GetSubject() + " (" + GetStartDate().Value.ToLocalTime() + ")";
-                }
+                string priority = Util.GetValueOfString(DB.ExecuteScalar(priorityValue));
+                type = "T";
+                title = Msg.GetMsg(GetCtx(), "Task");
+                body = GetSubject() + " (" + priority + ")";
+            }
+            else
+            {
+                type = "A";
+                title = Msg.GetMsg(GetCtx(), "Appointment");
+                body = GetSubject() + " (" + GetStartDate().Value.ToLocalTime() + ")";
+            }
 
+            if (newRecord)
+            {
                 PushNotification.SendNotificationToUser(GetAD_User_ID(), GetAD_Window_ID(), GetRecord_ID(), title, body, type);
             }
+            // Checking if notification is read by user 
+            else if (IsRead())
```

Comment:
Wrong check, need to check if the value is changed in IsRead column 

because in this case it will always fire if there is any change in some other column as well